MaterialLib/SolidModels/Ehlers-impl.h

```diff
@@ -464,6 +464,50 @@ void calculatePlasticJacobian(
     // G_52, G_53, G_55 are zero
 }
 
+template <int DisplacementDim>
+void SolidEhlers<DisplacementDim>::updateDamage(
+    double const t, ProcessLib::SpatialPosition const& x,
+    typename MechanicsBase<DisplacementDim>::MaterialStateVariables&
+        material_state_variables)
+{
+    assert(dynamic_cast<MaterialStateVariables*>(&material_state_variables) !=
+           nullptr);
+    MaterialStateVariables& _state =
+        static_cast<MaterialStateVariables&>(material_state_variables);
+
+    // Default case of the rate problem. Updated below if volumetric plastic
+    // strain rate is positive (dilatancy).
+    _state.kappa_d = _state.kappa_d_prev;
+
+    // Compute damage current step
+    double const eps_p_V_dot = _state.eps_p_V - _state.eps_p_V_prev;
+    if (eps_p_V_dot > 0)
+    {
+        double const h_d = _damage_properties->h_d(t, x)[0];
+        double const eps_p_eff_dot = _state.eps_p_eff - _state.eps_p_eff_prev;
+        double const r_s = eps_p_eff_dot / eps_p_V_dot;
+
+        // Brittleness decrease with confinement for the nonlinear flow rule.
+        // ATTENTION: For linear flow rule -> constant brittleness.
+        double x_s = 0;
+        if (r_s < 1)
+        {
+            x_s = 1 + h_d * r_s * r_s;
+        }
+        else
+        {
+            x_s = 1 - 3 * h_d + 4 * h_d * std::sqrt(r_s);
+        }
+        _state.kappa_d += eps_p_eff_dot / x_s;
```

Comment:
h_d is an input parameter and should not be smaller than one. It is technically possible to have x_s=0, but it will be because of wrong input.